cdap-proto/src/main/java/io/cdap/cdap/proto/TriggeringPipelineId.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto;
+
+import java.util.Objects;
+
+/**
+ * Class for identifiers of properties from the triggering pipeline.
+ */
+public class TriggeringPipelineId {
```

Comment:
there should not be anything called a 'pipeline' at the CDAP layer. CDAP doesn't know what a pipeline is, it only knows applications, programs, etc.